src/app/services/push.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable } from '@angular/core';
+import {
+	HttpClient,
+	HttpHeaders,
+	HttpParams
+} from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { ApiResponse, PushApiResponse } from '../models';
+@Injectable({
+	providedIn: 'root'
+})
+/*
+* The PushService is called on SEARCH_COMPLETE_SUCCESS
+* to push the result data back to server in the same
+* search result object format obtained from server.
+**/
+export class PushService {
+
+	constructor( private http: HttpClient ) { }
+
+	// Returns an Observable of PushApiResponse.
+	public postData(data: ApiResponse): Observable<PushApiResponse> {
+
+		// End point to make a Post to.
+		const httpUrl = 'https://api.loklak.org/api/push.json';
+
+		const headers = new HttpHeaders({
+			'Content-Type': 'application/x-www-form-urlencoded',
+			'Accept': 'application/json',
+			'cache-control': 'no-cache'
+		});
+		// Extracting search_metadata and statuses
+		// from the data obtained from search result.
+		const {search_metadata, statuses} = data;
```

Comment:
We're using `lowerCamelCase`, please be consistent.

Not sure why the build isn't failing for this.

Reply:
But ```search_metadata``` is the property inside data.

Reply:
```js
const { statuses } = data;
const searchMetadata = data.search_metadata;
```

This becomes long, but consistency is maintained. We can't control `search_metadata` in response, but we can control what we want to use as a variable.